test/src/tests-store.ts

```diff
@@ -1037,13 +1029,39 @@ describe('Store', () => {
             );
 
             store.setFirstSeenTime(testMPID, 54321);
+
+            expect(store.getFirstSeenTime(testMPID)).to.equal(12345);
+
             const fromPersistence = window.mParticle
                 .getInstance()
                 ._Persistence.getPersistence();
 
             expect(fromPersistence[testMPID]).to.be.ok;
             expect(fromPersistence[testMPID].fst).to.be.ok;
-            expect(fromPersistence[testMPID].fst).to.equal(54321);
+            expect(fromPersistence[testMPID].fst).to.equal(12345);
+        });
+
+        it('should set firstSeenTime once', () => {
+            const store: IStore = new Store(
+                sampleConfig,
+                window.mParticle.getInstance()
+            );
+
+            store.persistenceData['previous-set-mpid'] = {
+                fst: 100
+            }
```

Comment:
is there a reason you are setting fst for `previous-set-mpid` this way directly on `store.persistenceData` instead of how you are setting it for `current-mpid` and `previous-mpid` below?

Reply:
that sounds fine to me